### DISCUSSION TITLE
Architecture diagrams - custom icons

### DISCUSSION THREAD

**Discussion by PaulRudin** [mermaid-js/mermaid#5959] [General]
https://mermaid.js.org/syntax/architecture.html says that you can use custom icons, but I don't quite follow the instructions.
For example google publish icons for gcp : https://cloud.google.com/icons/files/google-cloud-icons.zip - what concrete steps would I take to generate a mermaid architecture diagram using these icons?

**Comment by Aaron-von-Awesome** [comment]
I have been at this all day, and still can't seem to figure it out :cry:  Any advice would be appreciated :innocent:

**Comment by ismailmmd** [comment]
Anyone got this figured out 🙏

**Comment by rbanffy** [comment]
As an alternative, would it be a good idea to add a basic set of icons to the architecture for some very common cases? We already have Disk, that cover things like NFS mounts and, perhaps, blob storage such as S3, and Database", which covers relational and non-relational persistent stores. There is nothing, however, to represent queues. Caches can be represented as a database, but using that icon might mislead, as caches are usually designed to be volatile.

I'll try my hand at adding the queue one to packages/mermaid/src/diagrams/architecture/architectureIcons.ts on a fork.

**Comment by hakanson** [comment]
I generated a set of the AWS Icons as part of https://github.com/awslabs/aws-icons-for-plantuml

If you want to see some code, start with https://github.com/awslabs/aws-icons-for-plantuml/blob/main/scripts/icon-builder.py#L892-L921 where I set `mermaid` to the basic iconfiy JSON structure.  The code below that calls `build_mermaid_icon` for each icon, and dumps a JSON file in the end.

Usage examples under https://github.com/awslabs/aws-icons-for-plantuml/tree/main/examples/mermaid-js
